pdata/pcommon/map_test.go

```diff
@@ -564,3 +564,29 @@ func TestInvalidMap(t *testing.T) {
 	assert.Panics(t, func() { v.AsRaw() })
 	assert.Panics(t, func() { _ = v.FromRaw(map[string]any{"foo": "bar"}) })
 }
+
+func TestMapEqual(t *testing.T) {
+	m := NewMap()
+	m2 := NewMap()
+	assert.True(t, m.Equal(m2))
+
+	m.PutStr("hello", "world")
+	assert.False(t, m.Equal(m2))
+
+	m2.PutStr("hello", "world")
+	assert.True(t, m.Equal(m2))
```

Comment:
Add a test for when both have "hello" but different values. Also a test when both have same len but different keys.